scalding-core/src/main/scala/com/twitter/scalding/Stats.scala

```diff
@@ -125,7 +125,7 @@ object RuntimeStats extends java.io.Serializable {
   def addFlowProcess(fp: FlowProcess[_]) {
     val uniqueJobIdObj = fp.getProperty(UniqueID.UNIQUE_JOB_ID)
     if (uniqueJobIdObj != null) {
-      uniqueJobIdObj.asInstanceOf[String].split(",").foreach { uniqueId =>
+      StringUtility.fastSplit(uniqueJobIdObj.asInstanceOf[String], ",").foreach { uniqueId =>
```

Comment:
this method will be called on every key in a group now (to fix a bug in counters inside of joins).

so, we really need to optimize this method.

1) let's not use .foreach. We should write a while loop.
2) we should check if flowMappingStore has the key already. If it does, there is no need to allocate a WeakReference.